### ISSUE TITLE
Error in SlintPad doesn't go away

### ISSUE THREAD

**Issue by Enyium** [slint-ui/slint#5949]
Go to https://slintpad.com/ and paste the following code:

```slint
component Bar {
    width: 100%;
    height: 100%;
    Rectangle { background: lime; }
}

export component Foo {
    Bar {}
}
```

As soon as you click on "Show Preview" for component `Bar`, errors are reported for both occurrences of `100`. When you then click on "Show Preview" for component `Foo`, the error doesn't go away, even though it shouldn't apply anymore. It only goes away when, after showing `Foo`'s preview, you completely remove (Ctrl+X) and paste the code again.

**Comment by ogoffart** [comment]
Thanks for the bug report.

I can reproduce it with slintpad.com which is on Slint 1.7.2, 
But I can no longer reproduce in the master branch: https://snapshots.slint.dev/master/editor/

So this appears to be fixed.

**Comment by Enyium** [comment]
Right, the master branch SlintPad you linked doesn't have this error. However, there's the following succession of previews:

After first pasting the code:

![1](https://github.com/user-attachments/assets/64a8122f-4108-463e-83f3-c5f455632663)

After pressing "Show Preview" on `Bar` and then on `Foo`:

![2](https://github.com/user-attachments/assets/4fe4b5ba-b541-42a1-a8ce-99c97856794c)

After again pressing "Show Preview" on `Foo`:

![3](https://github.com/user-attachments/assets/f49dc06e-d7d0-44df-a6f7-9fd249fd3017)

**Comment by ogoffart** [comment]
This seems to be a different issue. I've opened #5954 based on it.
